library/src/main/scala/scala/scalajs/js/JSArrayOps.scala

```diff
@@ -27,6 +27,12 @@ import scala.scalajs.js.annotation._
  *  To enable the use of these functions on js.[[Array]]s, import the implicit
  *  conversion [[JSArrayOps.jsArrayOps]].
  */
+@deprecated(
+    "It almost never makes sense to call native JavaScript Array functions " +
```

Comment:
In what cases does it make sense?

Reply:
For example if you have a sparse array, and you want holes to be skipped. But to get sparse JS arrays in Scala.js you have to mean it.

Reply:
Fair enough :)